### DISCUSSION TITLE
Error with dynamical url ?

### DISCUSSION THREAD

**Discussion by DamienA24** [vercel/next.js#18018] [Help]
Hello,

I have a project like this : 

> 
> src
> --pages
> ----website
> ------[id]
>--------file1.js
>--------file2.js

In my _app.js, I use getInitialProps for fetch my data.

When the id is good in my url no problem haha or if there are characters not provided  like "&", "(", "$" no problem also nextjs manage the errors.

But if the character is specific "%" I have a error by nextjs that I can't manage, the error is : 
![Capture d’écran 2020-10-19 à 16 09 36](https://user-images.githubusercontent.com/26389689/96462303-b5e97d00-1225-11eb-971f-0a2aa1993cc2.png)

![Capture d’écran 2020-10-19 à 16 10 11](https://user-images.githubusercontent.com/26389689/96462314-ba159a80-1225-11eb-8bad-f220aef8d885.png)

what is wrong ? why this character broke all ?

Thank you

**Comment by alanqchen** [comment]
The server will not be able to serve the page since the "%" is a [special escape character](https://en.wikipedia.org/wiki/Percent-encoding), and the server will decode the hex-encoded value that follows. For example, if you goto /cafe%20-restaurant, the server should be able to serve the page with no issues. However, in the URL you used, /cafe%-restaurant, "-r" is not a valid hex value and thus cannot be decoded.
